### ISSUE TITLE
Do you have any plan to publish ESPTool as a library?

### ISSUE THREAD

**Issue by mtsmfm** [xingrz/web-esptool#2]
Hello, thank you for sharing awesome codes.

I'm looking for a way to flash and monitor ESP on browser.

I can find a few implementations:

https://github.com/toitware/esptool.js
https://github.com/espressif/esptool-js

But I think src/esptool on web-esptool is the most well implemented one.

Do you have any chance to publish this src/esptool as a library on npmjs.com?

On my fork repo, I added some changes to make it publishable.

https://github.com/mtsmfm/web-esptool

If you'd like to do so, hopefully it helps.

**Comment by xingrz** [comment]
Of course. Technically `./src/esptool` is well-decoupled from the frontend, so it won't take me much extra work publishing to NPM.

**Comment by mtsmfm** [comment]
@xingrz I've added `read_flash` to `./src/esptool` on my fork.

https://github.com/mtsmfm/web-esptool/commit/87435f1437ffc1b25cfe44964f654d4120f91c9d

I'll send a PR to the library repo when you publish as a library.
